src/cpp/src/storage/checkpointer.cpp

```diff
@@ -15,6 +16,54 @@ Checkpointer::Checkpointer(std::shared_ptr<Model> model, shared_ptr<GraphModelSt
     config_ = config;
 }
 
+void Checkpointer::create_checkpoint(string checkpoint_dir, CheckpointMeta checkpoint_meta, int epochs, int frequency) {
+    int prev_checkpoint = epochs - frequency;
+    std::string prev_checkpoint_dir = checkpoint_dir + "checkpoint_" + std::to_string(prev_checkpoint) + "/";
+
+    string tmp_checkpoint_dir = checkpoint_dir + "checkpoint_" + std::to_string(epochs) + "_tmp/";
+    std::filesystem::create_directory(tmp_checkpoint_dir);
+    
+    std::string new_embeddings_file = tmp_checkpoint_dir
+                                        + PathConstants::embeddings_file 
+                                        + PathConstants::file_ext;
+    std::string new_embeddings_state_file = tmp_checkpoint_dir
+                                            + PathConstants::embeddings_state_file
+                                            + PathConstants::file_ext;
+
+    if(std::filesystem::exists(prev_checkpoint_dir)) {
```

Comment:
Why is this code needed? The embeddings/state from the previous checkpoint are moved to the new checkpoint before being overwritten by `this-save()`? 

Am I missing something or should this whole if/else be deleted.

Reply:
`this->save()` doesn't save the params to the checkpoint dir, it saves them to the `model_dir` i guess.

In this function `Checkpointer::save(...)`
`model->save()` function takes in a path argument and saves the model to the specified directory, while `storage_->storage_ptrs_.node_embeddings->write()` doesn't.